### ISSUE TITLE
Location info missing from malformed map nested in set or map-key position

### ISSUE THREAD

**Issue by yuhan0** [clj-kondo/clj-kondo#2811]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version** clj-kondo v2026.04.15

**macro usage** NA

**platform** native

**editor** NA

**problem**

Line and column numbers are missing from the linting output when an invalid map syntax (odd number of keys) occurs in a literal set.

**repro**
```clojure
(ns repro)

#{{:oops}}
```

```bash
$ clj-kondo --lint repro.clj
repro.clj:0:0: error: Can't parse repro.clj, No value supplied for key: :oops
linting took 45ms, errors: 1, warnings: 0
```

The same issue occurs wherever the invalid map is nested in the set, or appears in the key position of an enclosing map:

```clojure
;; Reports repro.clj:0:0 
#{arbitrary (amount [of nesting {:oops}]) levels}
{{:badkey} 123} 

;; OK (reports correct location)
{123 {:badval}}
[no #(parent (sets or {:oops}) maps) involved]
```

I traced it to this call to `(apply hash-map)` in  `clj-kondo.impl.linters.keys/key-value`
https://github.com/clj-kondo/clj-kondo/blob/eac5314853e291be6f3e00b24f69d4a1dde4f1b0/src/clj_kondo/impl/linters/keys.clj#L32
which throws an IllegalArgumentException that only gets caught at the top level handler. (also causes all subsequent linting errors in the file to be suppressed)

**expected behavior**

The usual `:missing-map-value` error message should be produced:
```
repro.clj:3:4: error: missing value for key :oops
```

Happy to submit a PR for this - I believe the simplest fix is to wrap the call site in a `(when (even? (count children)) ... (apply hash-map))`
unless there's a better place for such a guard

**Comment by borkdude** [comment]
Thanks. Fix underway.
